src/kinetics/Reaction.cpp

```diff
@@ -858,6 +877,46 @@ BlowersMasel readBlowersMasel(const Reaction& R, const AnyValue& rate,
     return BlowersMasel(A, b, Ta0, w);
 }
 
+bool detectEfficiencies(ThreeBodyReaction& R)
+{
+    for (const auto& reac : R.reactants) {
+        // detect explicitly specified collision partner
+        if (R.products.count(reac.first)) {
+            R.third_body.efficiencies[reac.first] = 1.;
+        }
+    }
+
+    if (R.third_body.efficiencies.size() == 0) {
+        return false;
+    } else if (R.third_body.efficiencies.size() > 1) {
+        throw CanteraError("detectEfficiencies",
+            "Found more than one explicitly specified collision partner\n"
+            "in reaction '{}'.", R.equation());
+    }
+
+    R.third_body.default_efficiency = 0.;
+    R.specified = true;
+    auto sp = R.third_body.efficiencies.begin();
+
+    // adjust reactant coefficients
+    auto reac = R.reactants.find(sp->first);
+    if (abs(reac->second - 1) > SmallNumber) {
```

Comment:
I get a compiler warning about this: `warning: using integer absolute value function 'abs' when argument is of floating point type [-Wabsolute-value]`. I think this (and the below instance) should be `std::abs`.

Reply:
Replaced by `trunc(reac->second) == 1`, which may be clearer anyways.